consensus/src/epoch_manager.rs

```diff
@@ -1512,6 +1513,44 @@ impl<P: OnChainConfigProvider> EpochManager<P> {
         Ok(())
     }
 
+    fn check_author(&mut self, peer_id: AccountAddress, msg: &ConsensusMsg) -> anyhow::Result<()> {
+        let author = match msg {
+            ConsensusMsg::CommitMessage(commit) => commit.author(),
+            ConsensusMsg::ProposalMsg(proposal) => proposal.proposal().author(),
+            ConsensusMsg::VoteMsg(vote) => Some(vote.vote().author()),
+            ConsensusMsg::OrderVoteMsg(order_vote) => Some(order_vote.order_vote().author()),
+            ConsensusMsg::CommitVoteMsg(commit_vote) => Some(commit_vote.author()),
+            ConsensusMsg::BatchMsg(batch) => batch.author(),
+            ConsensusMsg::RoundTimeoutMsg(round_timeout) => Some(round_timeout.author()),
+            ConsensusMsg::BatchResponse(batch_response) => Some(batch_response.author()),
```

Comment:
I don't think this works too, even this doesn't go through this path (it goes through rpc), this is the batch author which is not necessarily the responser.